packages/docusaurus-plugin-content-docs/src/props.ts

```diff
@@ -224,3 +225,18 @@ export function toTagsListTagsProp(
       count: tagValue.docIds.length,
     }));
 }
+
+export function calculateLastModified({
+  frontMatter,
+  lastUpdatedAt,
+}: DocMetadata): Date | undefined {
+  const frontMatterLastUpdate = frontMatter?.last_update?.date;
+  if (frontMatterLastUpdate !== undefined) {
+    return typeof frontMatterLastUpdate === 'string'
+      ? new Date(frontMatterLastUpdate)
+      : frontMatterLastUpdate;
+  } else if (lastUpdatedAt !== undefined) {
+    return new Date(lastUpdatedAt * 1000);
+  }
+  return undefined;
+}
```

Comment:
front matter already overrides `doc.lastUpdatedAt` value so this code is probably not needed, and you can use `lastUpdatedAt` directly

The only problem is that `lastUpdatedAt` will not be set if `showLastUpdateTime: false` (default), and this attribute wasn't really created to have an impact on sitemap lastmod so it's a bit confusing.